showcase/public/ellipsizable-text.js

```diff
@@ -0,0 +1,97 @@
+const template = Object.assign(document.createElement('template'), {
```

Comment:
So, I was doing some tests here because I noticed that we are adding the ellipsis class only when we know that the text should overflow, this generates a redraw. we can set the ellipsis attributes by default and the just calculate if we should show the tooltip or not, this way we avoid the redraw right?

Reply:
I'm not sure we can avoid the redraw, it happens because I first render just the plain text letting it overflow and **then** I truncate it. If we skip the overflow then we can't detect it because the text's width will always match its parent. The only alternatives to detect the overflow without using `offsetWidth` are either using a canvas with `measureText` (which I'm pretty sure is slower) and using a monospaced font for the table so we can count the cols.